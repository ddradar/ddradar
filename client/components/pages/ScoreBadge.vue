<template>
  <b-tooltip :label="clearStatus">
    <span
      class="tag is-light"
      :class="{
        'is-white': lamp === 7,
        'is-basic': lamp === 6,
        'is-expert': lamp === 5,
        'is-beginner': lamp === 4,
        'is-difficult': lamp === 3,
        'is-challenge': lamp === 1,
        'is-light': lamp === 0,
      }"
    >
      {{ score }}
    </span>
  </b-tooltip>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator'

import { ClearLamp, clearLampList } from '~/types/api/score'

@Component
export default class ScoreBadgeComponent extends Vue {
  @Prop({ required: true, type: Number })
  readonly lamp!: ClearLamp

  @Prop({ required: true, type: Number })
  readonly score!: number

  get clearStatus() {
    return clearLampList[this.lamp]
  }
}
</script>
