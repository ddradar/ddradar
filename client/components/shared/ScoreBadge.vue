<template>
  <b-tooltip :label="clearStatus">
    <span
      class="tag is-light"
      :class="{
        'is-white': lamp === 7,
        'is-basic': lamp === 6,
        'is-expert': lamp === 5,
        'is-beginner': lamp === 4,
        'is-difficult': lamp === 3,
        'is-challenge': lamp === 1,
        'is-light': lamp === 0,
      }"
    >
      {{ score }}
    </span>
  </b-tooltip>
</template>

<script lang="ts">
import { Score } from '@ddradar/core'
import { Component, Prop, Vue } from 'nuxt-property-decorator'

@Component
export default class ScoreBadgeComponent extends Vue {
  @Prop({ required: true, type: Number })
  readonly lamp!: Score.ClearLamp

  @Prop({ required: true, type: Number })
  readonly score!: number

  get clearStatus() {
    return Score.clearLampMap.get(this.lamp)
  }
}
</script>
