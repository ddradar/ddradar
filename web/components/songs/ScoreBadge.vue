<template>
  <OTooltip :label="clearLampMap.get(lamp)">
    <span class="tag" :class="classMap.get(lamp)">{{ score }}</span>
  </OTooltip>
</template>

<script lang="ts" setup>
import type { Score } from '@ddradar/core'
import { clearLampMap } from '@ddradar/core'

interface ScorebadgeProps {
  lamp: Score['clearLamp']
  score: Score['score']
}
const classMap = new Map([
  [0, 'is-dark'],
  [1, 'is-challenge'],
  [2, 'is-primary'],
  [3, 'is-difficult'],
  [4, 'is-beginner'],
  [5, 'is-expert'],
  [6, 'is-basic'],
  [7, 'is-white'],
] as const)

defineProps<ScorebadgeProps>()
</script>
