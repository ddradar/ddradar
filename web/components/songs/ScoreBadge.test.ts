import Oruga from '@oruga-ui/oruga-next'
import { bulmaConfig } from '@oruga-ui/theme-bulma'
import { mount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'

import ScoreBadge from '~~/components/songs/ScoreBadge.vue'

describe('components/songs/ScoreBadge.vue', () => {
  test.each([
    [0, 10000],
    [1, 340000],
    [2, 800000],
    [3, 950000],
    [4, 940000],
    [5, 987650],
    [6, 999990],
    [7, 1000000],
  ])('{ lamp: %i, score: %i } renders correctly', (lamp, score) => {
    const wrapper = mount(ScoreBadge, {
      props: { lamp, score },
      global: { plugins: [[Oruga, bulmaConfig]] },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
