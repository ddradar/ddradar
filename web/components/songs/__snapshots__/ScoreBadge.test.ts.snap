// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/songs/ScoreBadge.vue > { lamp: 0, score: 10000 } renders correctly 1`] = `
<div
  class="b-tooltip is-primary is-top"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      class="tooltip-content is-top"
      style="display: none;"
    >
      <span
        class=""
      />
      
      Failed
      
    </div>
  </transition-stub>
  <div
    class="tooltip-trigger"
  >
    
    <span
      class="tag is-dark"
    >
      10000
    </span>
    
  </div>
</div>
`;

exports[`components/songs/ScoreBadge.vue > { lamp: 1, score: 340000 } renders correctly 1`] = `
<div
  class="b-tooltip is-primary is-top"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      class="tooltip-content is-top"
      style="display: none;"
    >
      <span
        class=""
      />
      
      Assisted Clear
      
    </div>
  </transition-stub>
  <div
    class="tooltip-trigger"
  >
    
    <span
      class="tag is-challenge"
    >
      340000
    </span>
    
  </div>
</div>
`;

exports[`components/songs/ScoreBadge.vue > { lamp: 2, score: 800000 } renders correctly 1`] = `
<div
  class="b-tooltip is-primary is-top"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      class="tooltip-content is-top"
      style="display: none;"
    >
      <span
        class=""
      />
      
      Clear
      
    </div>
  </transition-stub>
  <div
    class="tooltip-trigger"
  >
    
    <span
      class="tag is-primary"
    >
      800000
    </span>
    
  </div>
</div>
`;

exports[`components/songs/ScoreBadge.vue > { lamp: 3, score: 950000 } renders correctly 1`] = `
<div
  class="b-tooltip is-primary is-top"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      class="tooltip-content is-top"
      style="display: none;"
    >
      <span
        class=""
      />
      
      Life 4
      
    </div>
  </transition-stub>
  <div
    class="tooltip-trigger"
  >
    
    <span
      class="tag is-difficult"
    >
      950000
    </span>
    
  </div>
</div>
`;

exports[`components/songs/ScoreBadge.vue > { lamp: 4, score: 940000 } renders correctly 1`] = `
<div
  class="b-tooltip is-primary is-top"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      class="tooltip-content is-top"
      style="display: none;"
    >
      <span
        class=""
      />
      
      Full Combo
      
    </div>
  </transition-stub>
  <div
    class="tooltip-trigger"
  >
    
    <span
      class="tag is-beginner"
    >
      940000
    </span>
    
  </div>
</div>
`;

exports[`components/songs/ScoreBadge.vue > { lamp: 5, score: 987650 } renders correctly 1`] = `
<div
  class="b-tooltip is-primary is-top"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      class="tooltip-content is-top"
      style="display: none;"
    >
      <span
        class=""
      />
      
      Great Full Combo
      
    </div>
  </transition-stub>
  <div
    class="tooltip-trigger"
  >
    
    <span
      class="tag is-expert"
    >
      987650
    </span>
    
  </div>
</div>
`;

exports[`components/songs/ScoreBadge.vue > { lamp: 6, score: 999990 } renders correctly 1`] = `
<div
  class="b-tooltip is-primary is-top"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      class="tooltip-content is-top"
      style="display: none;"
    >
      <span
        class=""
      />
      
      Perfect Full Combo
      
    </div>
  </transition-stub>
  <div
    class="tooltip-trigger"
  >
    
    <span
      class="tag is-basic"
    >
      999990
    </span>
    
  </div>
</div>
`;

exports[`components/songs/ScoreBadge.vue > { lamp: 7, score: 1000000 } renders correctly 1`] = `
<div
  class="b-tooltip is-primary is-top"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      class="tooltip-content is-top"
      style="display: none;"
    >
      <span
        class=""
      />
      
      Marvelous Full Combo
      
    </div>
  </transition-stub>
  <div
    class="tooltip-trigger"
  >
    
    <span
      class="tag is-white"
    >
      1000000
    </span>
    
  </div>
</div>
`;
